lib/logstash/filters/math.rb

```diff
@@ -108,18 +112,18 @@ def register
 
   def filter(event)
     event_changed = false # can exit if none of the calculations are are suitable
-    @register.clear # don't carry over register results from one event to the next.
+    store = Storage.new(event) # creates a new registers array, each element can use the event or register from the store
```

Comment:
I am not sure I like the `Storage` extra abstraction ... after re-reading code I see the `MathCalculationElements` idea of the same get/set interface for the subclasses ... so the `Storage` in the line of though makes sense I guess.   I personally would have revisited this whole abstraction, seems to complicate comprehension more that simplify it for something already relatively simple IMO - but that's a five minutes assertion :P

Reply:
Naming things is always one of the hardest parts, but `Storage` to me doesn't signify that we're dealing with something that has an event and an event's register in it.

Names that may more effectively encapsulate what it is:
 - `EventRegister`
 - `EventStore`
 - `EventContext`